HW18/homework_18.py

```diff
@@ -0,0 +1,67 @@
+class Bot:
+
+    def __init__(self, name):
+        self.name = name
+
+    def say_name(self):
+        print(self.name)
+
+    def send_message(self, message):
+        print(message)
+
+
+class TelegramBot(Bot):
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.chat_id = None
+        self.url = None
+
+    def set_url(self, url=None):
+        self.url = url
+
+    def set_chat_id(self, chat_id=None):
+        self.chat_id = chat_id
+
+    def send_message(self, message):
+        print(f"{self.name} bot says {message} to chat {self.chat_id} using url {self.set_url()}")
```

Comment:
self.set_url() will reset your url even if you set it before and won't return any value to print. Maybe you meant `self.url`?

Reply:
yes, you're right. I meant self.url